cloudinit/apport.py

```diff
@@ -101,8 +102,25 @@ def attach_hwinfo(report, ui=None):
 
 
 def attach_cloud_info(report, ui=None):
-    """Prompt for cloud details if available."""
+    """Prompt for cloud details if instance-data unavailable.
+
+    When we have valid _get_instance_data, apport/generic-hooks/cloud_init.py
```

Comment:
Where does this `apport/generic_hooks/cloud_init.py` live?

Reply:
This is delivered by our downstream packaging branches per the (yet to be landed) https://github.com/canonical/cloud-init/pull/2020 from @bdrung . We'll reflect the generic hooks delivery into each ubuntu/* packaging branch. Since apport is only ubuntu, we don't really have to cope with this in other distributions. Ideally, I'd like to move even this apport integration down into packaging branches too. But, not for this PR

Reply:
Also note, we will reflect #2020 to each ubuntu/(bionic|focal|jammy|kinetic) once this PR and 2020 land

Reply:
Makes sense. I think that could be a little more clear in the docstring (or comment near the change) as it's a little strange having a function obtain a value and then not do anything with it.